src/component/axis/AxisBuilder.ts

```diff
@@ -347,6 +348,20 @@ const builders: Record<'axisLine' | 'axisTickLabel' | 'axisName', AxisElementsBu
         fixMinMaxLabelShow(axisModel, labelEls, ticksEls);
 
         buildAxisMinorTicks(group, transformGroup, axisModel, opt.tickDirection);
+
+        // This bit fixes the label overlap issue for the Time chart.
+        // See https://github.com/apache/echarts/issues/14266 for more.
+        if (axisModel.type === 'xAxis.time') {
```

Comment:
It will be slightly better to do the check by `axisModel.get('type') === 'time'`. Because this axis may be `yAxis` if the layout is vertical.